error when compiling

before new update CRSFJoystick all work fine. but now I download new archive and can't compile. pls help.

<img width="1217" height="381" alt="Image" src="https://github.com/user-attachments/assets/4a2c734d-21a9-4bb8-92e5-445aa9a2db8e" />

`.pio\libdeps\waveshare2040zero\Adafruit NeoPixel\Adafruit_Neopixel_RP2.cpp: In member function 'bool Adafruit_NeoPixel::rp2040claimPIO()':
.pio\libdeps\waveshare2040zero\Adafruit NeoPixel\Adafruit_Neopixel_RP2.cpp:9:9: error: 'pio_claim_free_sm_and_add_program_for_gpio_range' was not declared in this scope
    9 |   if (! pio_claim_free_sm_and_add_program_for_gpio_range(&ws2812_program,
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.pio\libdeps\waveshare2040zero\Adafruit NeoPixel\Adafruit_Neopixel_RP2.cpp: In member function 'void Adafruit_NeoPixel::rp2040releasePIO()':
.pio\libdeps\waveshare2040zero\Adafruit NeoPixel\Adafruit_Neopixel_RP2.cpp:35:3: error: 'pio_remove_program_and_unclaim_sm' was not declared in this scope
   35 |   pio_remove_program_and_unclaim_sm(&ws2812_program, pio, pio_sm,  pio_program_offset);
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.pio\libdeps\waveshare2040zero\Adafruit NeoPixel\Adafruit_Neopixel_RP2.cpp: In member function 'void Adafruit_NeoPixel::rp2040Show(uint8_t*, uint32_t)':
.pio\libdeps\waveshare2040zero\Adafruit NeoPixel\Adafruit_Neopixel_RP2.cpp:43:24: warning: comparison of unsigned expression < 0 is always false [-Wtype-limits]
   43 |   if (! pio || (pio_sm < 0)) {
      |                 ~~~~~~~^~~
*** [.pio\build\waveshare2040zero\libf3a\Adafruit NeoPixel\Adafruit_Neopixel_RP2.cpp.o] Error 1`

--- a/include/board_defs.h
+++ b/include/board_defs.h
@@ -63,26 +63,23 @@
     #define LED_TYPE LED_TYPE_NEOPIXEL
     #define LED_PIN 16
 
-    #include <Adafruit_NeoPixel.h>
-    Adafruit_NeoPixel pixels(1, LED_PIN, NEO_GRB + NEO_KHZ800);
+    #include <NeoPixelConnect.h>
+    NeoPixelConnect p(LED_PIN, 1, pio0, 0);
     
     void boardSetup() {
       // noop
     }
 
     void led_off() {
-      pixels.setPixelColor(0, pixels.Color(0,0,0));
-      pixels.show();
+        p.neoPixelFill(0, 0, 0, true);
     }
     
     void led_on() {
-      pixels.setPixelColor(0, pixels.Color(0,0,255));
-      pixels.show();
+        p.neoPixelFill(0, 0, 255, true);
     }
 
     void led_color(uint8_t r, uint8_t g, uint8_t b) {
-      pixels.setPixelColor(0, pixels.Color(r,g,b));
-      pixels.show();
+        p.neoPixelFill(r, g, b, true);
     }
 #endif
 
@@ -96,8 +93,8 @@
     #define LED_NEO_PWR 11
     #define LED_PIN 12
 
-    #include <Adafruit_NeoPixel.h>
-    Adafruit_NeoPixel pixels(1, LED_PIN, NEO_GRB + NEO_KHZ800);
+    #include <NeoPixelConnect.h>
+    NeoPixelConnect p(LED_PIN, 1, pio0, 0);
     
     // 1 Active Low RGB
     //#define LED_ACTIVE_LOW LED_ACTIVE_LOW_TRUE
@@ -120,14 +117,16 @@
     }
 
     void led_off() {
-      pixels.setPixelColor(0, pixels.Color(0,0,0));
-      pixels.show();
+        p.neoPixelFill(0, 0, 0, true);
     }
     
     void led_on() {
-      pixels.setPixelColor(0, pixels.Color(0,0,255));
-      pixels.show();
-    }    
+        p.neoPixelFill(0, 0, 255, true);
+    }
+
+    void led_color(uint8_t r, uint8_t g, uint8_t b) {
+        p.neoPixelFill(r, g, b, true);
+    }
 #endif
 
 // Adafruit QT Py RP2040
@@ -140,8 +139,8 @@
     #define LED_NEO_PWR 11
     #define LED_PIN 12
 
-    #include <Adafruit_NeoPixel.h>
-    Adafruit_NeoPixel pixels(1, LED_PIN, NEO_GRB + NEO_KHZ800);
+    #include <NeoPixelConnect.h>
+    NeoPixelConnect p(LED_PIN, 1, pio0, 0);
 
     void boardSetup() {
         pinMode(LED_NEO_PWR, OUTPUT);
@@ -149,13 +148,15 @@
     }
 
     void led_off() {
-      pixels.setPixelColor(0, pixels.Color(0,0,0));
-      pixels.show();
+        p.neoPixelFill(0, 0, 0, true);
     }
     
     void led_on() {
-      pixels.setPixelColor(0, pixels.Color(0,0,255));
-      pixels.show();
+        p.neoPixelFill(0, 0, 255, true);
+    }
+
+    void led_color(uint8_t r, uint8_t g, uint8_t b) {
+        p.neoPixelFill(r, g, b, true);
     }
 #endif
 

--- a/platformio.ini
+++ b/platformio.ini
@@ -8,14 +8,11 @@
 ; Please visit documentation for the other options and examples
 ; https://docs.platformio.org/page/projectconf.html
 
-
 [env:pico]
 platform = raspberrypi
 board = pico
 framework = arduino
-lib_deps = 
-	fastled/FastLED@^3.6.0
-	adafruit/Adafruit NeoPixel@^1.11.0
+lib_deps = mryslab/NeoPixelConnect@^1.4.0
 build_flags = -DBOARD_ID=1
 monitor_speed = 115200
 
@@ -24,8 +21,7 @@ platform = raspberrypi
 board = pico
 framework = arduino
 lib_deps = 
-	fastled/FastLED@^3.6.0
-	adafruit/Adafruit NeoPixel@^1.11.0
+	mryslab/NeoPixelConnect@^1.4.0
 build_flags = -DBOARD_ID=2
 monitor_speed = 115200
 
@@ -33,9 +29,7 @@ monitor_speed = 115200
 platform = raspberrypi
 board = pico
 framework = arduino
-lib_deps = 
-	fastled/FastLED@^3.6.0
-	adafruit/Adafruit NeoPixel@^1.11.0
+lib_deps = mryslab/NeoPixelConnect@^1.4.0
 build_flags = -DBOARD_ID=3
 monitor_speed = 115200
 
@@ -44,8 +38,7 @@ platform = raspberrypi
 board = pico
 framework = arduino
 lib_deps = 
-	fastled/FastLED@^3.6.0
-	adafruit/Adafruit NeoPixel@^1.11.0
+	mryslab/NeoPixelConnect@^1.4.0
 build_flags = -DBOARD_ID=4
 monitor_speed = 115200
 
@@ -54,7 +47,6 @@ platform = raspberrypi
 board = pico
 framework = arduino
 lib_deps = 
-	fastled/FastLED@^3.6.0
-	adafruit/Adafruit NeoPixel@^1.11.0
+	mryslab/NeoPixelConnect@^1.4.0
 build_flags = -DBOARD_ID=5
 monitor_speed = 115200

